ai/shared-concepts.md

```diff
@@ -94,24 +145,45 @@ Your producer and consumer determine event ingestion and message submission:
 ---
 
 ## 5. Job Signature Conventions
-Use extractors to simplify job argument handling:
-
-- `TangleArg<T>`: one field
-- `TangleArgs2<A, B>`: two fields
-- `BlockEvents`: EVM logs
-- `Context<MyContext>`: context injection
 
-Return `TangleResult<T>` or `Result<(), Error>` depending on job type.
+### Handler Pattern
+Job handlers use extractors for context and argument handling:
 
 ```rust
-pub async fn handler(
+#[debug_job]  // Automatic entry/exit logging
+pub async fn handler_name(
     Context(ctx): Context<MyContext>,
     TangleArg(data): TangleArg<String>,
-) -> Result<TangleResult<u64>> {
-    ...
+    // Other extractors as needed
+) -> Result<TangleResult<U>, Error> {
+    // Implementation
 }
 ```
 
+**Key Components:**
+- **Extractors:** Use these to handle inputs automatically:
+  - `Context<MyContext>`: Injects the context
+  - `TangleArg<T>`: Single field argument
+  - `TangleArgs2<A, B>`, `TangleArgs3<A, B, C>`: Multiple fields
+  - `Optional<T>`: For optional arguments
+  - `List<T>`: For array/list arguments
+  - `BlockEvents`: For extracting EVM logs
```

Comment:
This is mixing tangle and evm extractors, they should be listed separately to avoid confusion.